### ISSUE TITLE
PsrStorageFactory references Psr\SimpleCache\CacheInterface;

### ISSUE THREAD

**Issue by maticb** [jeremykendall/php-domain-parser#360]
## Issue summary

So the file (Pdp\Storage\PsrStorageFactory )  references an interface which is a dependency (psr/simple-cache) only in require-dev, but it is not clear to me that this is for dev use, according to the documentation? Is this PsrStorageFactory not supposed to be used in a production enviroment, or is the composer require incorrectly under "dev"?

**Comment by nyamsprod** [comment]
Hi @maticb thanks for using the library. Indeed the class you are referring to depends on the availability of a PSR simple cache implementation which is not present in the `require` section of the `composer.json` the reason are the following:

To use the library you **do not require** any cache (even though it is highly recommended to use one).

The storage namespace provides interfaces and classes to enable caching the resource using any type of cache but also, in the case of the PSR simple cache, an extra layer is added to ease using any PSR simple cache implementation **BUT** the library stop from providing any concrete implementation. Different applications use different cache mechanisms and even different PSR simple cache implementation so we did not want to impose a specific package that might conflict with your application dependencies.

Last but not least, on installation, the package **does recommend installing cache related packages** using the `suggest` field from the `composer.json` which is the goal of that section of the manifest.

TL;DR:

The `composer.json` contains:

- the `require` section contains the minimum required package for the library to work
- the `require-dev` section is irrelevant for anyone consuming the package
- the `suggest` section contains information about extra packages that you may install if you need to do more with the package (but which is not a requirement for the package to work).

You are free to use any class inside the `Storage` namespace as long as you also install or provide classes that implement the required cache interfaces in your application `composer.json` manifest

**Comment by maticb** [comment]
Hmm, okay, I am unfamiliar with composer suggests, it seems odd to me that code would fail if you do not install another package, because it seems my symfony cache also did not include this simple-cache plugin.

But if this is the standard procedure, then okay. thank you for the reply :)

**Comment by maticb** [comment]
Yes, I used use Symfony\Component\Cache\Psr16Cache;, but weirdly I just noticed that also uses Psr\SimpleCache\CacheInterface which my project did not have in composer, I had to manually install that in composer.
